### ISSUE TITLE
create interop session(s)

### ISSUE THREAD

**Issue by petervanderstok** [anima-wg/constrained-voucher#104]
possiby during the coming IETF hackaton (July 2021)
Panos interested?

**Comment by csosto-pk** [comment]
Hi @petervanderstok , my team has not done any work on constrained voucher or join proxy so we would be able to contribute to the interop. They have been working on different things recently. Sorry.

**Comment by EskoDijk** [comment]
Note for the Registrar/EST-server implementations: my Pledge will request content-format application/pkix-cert (287) in its request to /.well-known/est/crts . This is defined as OPTIONAL to support by the EST-server in ace-coap-est, however from a practical point of view this needs to be supported otherwise it won't work.  (Assuming that Pledge use of 287 is a "MAY", it would follow that EST-server support for 287 is a "MUST".  That consideration did not make it into ace-coap-est unfortunately...)

**Comment by EskoDijk** [comment]
Question for the interop session: how do we plan to arrange the connectivity from Registrar to MASA server? For example the IDevID of the Pledge may encode a MASA URI "myserver.example.com:12345" .  Do we know that the VPN used during the interop session allows a Registrar to contact this URI/port over IPv4 or IPv6? Or are all the network interactions constrained to the local VPN only?  In case of local-only, what MASA URIs can we use in the IDevID ?

**Comment by petervanderstok** [comment]
I would say, that all connections and certificates need to be specified 
for interop.
I hope we can do interop with simulated MASA (possibly inside Registrar 
?)
Already doing the DTLS connection between pledge and Registrar (no join 
proxy to start with) may prove to be a challenge.
Esko Dijk schreef op 2021-04-26 10:47:

> Question for the interop session: how do we plan to arrange the 
> connectivity from Registrar to MASA server? For example the IDevID of 
> the Pledge may encode a MASA URI "myserver.example.com:12345" . Do we 
> know that the VPN used during the interop session allows a Registrar to 
> contact this URI/port over IPv4 or IPv6? Or are all the network 
> interactions constrained to the local VPN only? In case of local-only, 
> what MASA URIs can we use in the IDevID ?
> 
> --
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub [1], or unsubscribe 
> [2].


Links:
------
[1] 
https://github.com/anima-wg/constrained-voucher/issues/104#issuecomment-826639922
[2]

**Comment by mcr** [comment]
> Question for the interop session: how do we plan to arrange the connectivity from Registrar to MASA server? For example the IDevID of the Pledge may encode a MASA URI "myserver.example.com:12345" . Do we know that the VPN used during the interop session allows a Registrar to contact this URI/port over IPv4 or IPv6? Or are all the network interactions constrained to the local VPN only? In case of local-only, what MASA URIs can we use in the IDevID ?

The VPN that I have, based upon RPI, and IPsec, allows full connectivity in and out of the network.  They VPN was just virtual cables, simulating the regular IETF network.  You could easily put "myserver.example.com" with an IETF IPv4/v6 A/AAAA record in your DNS, and it would work.
We are promised some new system based upon mikrotik boxes and maybe wireguard, but aside from the hardware changes, I think the goal is the same.

However: I think that many will probably find it easier to just put their *MASA* up at some public address (EC2, linode, self-hosted).
What we need the "LAN" from the IETF network is to allow for discovery messages to work so that Pledges can find Join-Proxy/Registrars.

**Comment by mcr** [comment]
> I would say, that all connections and certificates need to be specified for 
> I hope we can do interop with simulated MASA (possibly inside Registrar ?) 

Please tell me more.
The only way I can see to simulate the MASA (to fake the voucher) is if you convince the pledge to use a different verification key, or skip the verification completely.
I have generally assumed that people who bring a pledge have to bring a MASA.  Since I know not everyone has a MASA, I put up two MASA that offer key pairs.

https://minerva.sandelman.ca/jokeshop/ explains what I did.
Basically: you grab a zip file with IDevID and matching private key and "provision" it to your device.   Once the MASA thinks that the device has got an owner, it takes the zip file away and makes a new one.  So, there is a race condition that two experimenters can grab the same one, but it's not a big deal.   The private keys being exposed, there is no real security, but we can pretend this part.

**Comment by EskoDijk** [comment]
For me also fine to put my MASA up on the public internet. I remember a problem I sometimes had (and others too) was getting a proper IPv6 connectivity from home to the server e.g. in EC2. IPv4 was then used as fallback.

**Comment by mcr** [comment]
https://docs.google.com/document/d/1T8Rtfk1zia_p05_6eb_WQA2Mmid-eP1-cAgnwdpF9Xk/edit?usp=sharing

**Comment by EskoDijk** [comment]
Closing the IETF111 hackathon interop ticket. Google Docs file will remain in use
